src/overlays/actors/ovl_Boss_06/z_boss_06.c

```diff
@@ -521,8 +520,8 @@ void Boss06_Draw(Actor* thisx, GlobalContext* globalCtx2) {
     func_8012C28C(globalCtx->state.gfxCtx);
 
     temp_v0 = gSaveContext.time;
-    if (gSaveContext.time > 0x8000) {
-        temp_v0 = (0xFFFF - gSaveContext.time) & 0xFFFF;
+    if (temp_v0 > 0x8000) {
+        temp_v0 = (0xFFFF - temp_v0);
     }
     sp68 = (f32)temp_v0 / 0x8000;
```

Comment:
i wonder if this is a macro or if it uses `(void)0`

Reply:
Not sure how `(void)0` would work, since this isn't just accessing the variable. Its more like a `CLAMP` macro. Though of note this same construct also exists in `EnKanban_Draw`, however it uses `>=` instead of `>` so couldn't be combined into a single macro.

Reply:
On this subject noticed a couple cleanup things that am going to add real quick. Namely removing the ` & 0xFFFF` from `EnKanban_Draw`, and using `CLOCK_TIME` in `Boss06_Draw`.

Reply:
Now that you mention Kanban... I think this is a common pattern from our guy Morita when he wants to make stuff that changes based on the time of day. Kanban and Jsjutan on OoT have it too
I doubt we would see this pattern much, and probably isn't worth to make a macro